core/src/main/java/edu/wpi/grip/core/operations/composite/CropOperation.java

```diff
@@ -87,14 +88,13 @@ public void perform() {
     final Origin origin = originSocket.getValue().get();
 
     final Rect regionOfInterest = new Rect(
-            x.intValue() + (int)(origin.xOffsetMultiplier * width.intValue()),
-            y.intValue() + (int)(origin.yOffsetMultiplier * height.intValue()),
+            x.intValue() + (int) (origin.xOffsetMultiplier * width.intValue()),
+            y.intValue() + (int) (origin.yOffsetMultiplier * height.intValue()),
             width.intValue(),
             height.intValue()
             );
 
-    final Mat output = new Mat(input, regionOfInterest).clone();
-
+    input.apply(regionOfInterest).copyTo(output);
```

Comment:
Does `apply` modify the input? If so, it might be more appropriate to do:
```
input.copyTo(output);
output.apply(regionOfInterest);
```

Reply:
`apply` doesn't modify the input. It just returns the specified submatrix of the original Mat.

![image](https://user-images.githubusercontent.com/11065157/54937920-63ce6180-4ef3-11e9-8b9c-d1dd8e8545bf.png)
I added a blur off of the input of the crop and it seems to work just fine.

Reply:
This is 100% the test I was thinking of having you run to ensure that this worked correctly. Thank you for ensuring this. Would you be so kind as to leave a comment explaining this so that 6 months from now when I look at this code I know this is intentional? A simple link to this thread and a comment explaining what you said above should be enough.